backend/src/main/java/ai/verta/modeldb/utils/ModelDBHibernateUtil.java

```diff
@@ -561,109 +562,37 @@ public static Query getWorkspaceEntityQuery(
-                    }
-                  }
-                  if (index > 0) {
-                    CompletableFuture<Void> combinedFuture =
-                        CompletableFuture.allOf(completableFutures);
-                    combinedFuture.get();
-                    LOGGER.info("Finished all the future tasks");
-                  }
-                } catch (InterruptedException | ExecutionException e) {
-                  LOGGER.warn(
-                      "ModelDBHibernateUtil runMigration() getting error : {}", e.getMessage(), e);
-                } finally {
-                  CommonUtils.unregisteredBackgroundUtilsCount();
-                }
-              })
-          .start();
-
-      // Blocking migration
-      String migrationName = ModelDBConstants.DATASET_VERSIONING_MIGRATION;
-      if (migrationTypeMap.containsKey(migrationName)) {
-        Map<String, Object> migrationDetailMap = migrationTypeMap.get(migrationName);
-        if ((boolean) migrationDetailMap.get(ModelDBConstants.ENABLE)) {
-          try {
+    if (config.migrations != null) {
+      for (MigrationConfig migrationConfig : config.migrations) {
+        switch (migrationConfig.name) {
+          case ModelDBConstants.SUB_ENTITIES_OWNERS_RBAC_MIGRATION:
+            OwnerRoleBindingUtils.execute();
+            break;
+          case ModelDBConstants.SUB_ENTITIES_REPOSITORY_OWNERS_RBAC_MIGRATION:
+            OwnerRoleBindingRepositoryUtils.execute();
+            break;
+          case ModelDBConstants.POPULATE_VERSION_MIGRATION:
+            PopulateVersionMigration.execute(migrationConfig.record_update_limit);
+            break;
+          case ModelDBConstants.DATASET_VERSIONING_MIGRATION:
             CommonUtils.registeredBackgroundUtilsCount();
-            boolean isLocked = checkMigrationLockedStatus(migrationName, rdb);
+            boolean isLocked = checkMigrationLockedStatus(migrationConfig.name, rdb);
             if (!isLocked) {
-              LOGGER.debug("Obtaingin migration lock");
-              lockedMigration(migrationName, rdb);
-              int recordUpdateLimit =
-                  (int) migrationDetailMap.getOrDefault(ModelDBConstants.RECORD_UPDATE_LIMIT, 100);
-              DatasetToRepositoryMigration.execute(recordUpdateLimit);
+              LOGGER.debug("Obtaining migration lock");
+              lockedMigration(migrationConfig.name, rdb);
+              DatasetToRepositoryMigration.execute(migrationConfig.record_update_limit);
             } else {
               LOGGER.debug("Migration already locked");
             }
-          } catch (SQLException | DatabaseException e) {
-            LOGGER.error("Error on migration: {}", e.getMessage());
-          } finally {
-            CommonUtils.unregisteredBackgroundUtilsCount();
-          }
+            break;
         }
       }
-
-      CollaboratorResourceMigration.execute();
     }
+
+    CollaboratorResourceMigration.execute();
```

Comment:
This was inside the dataset version migration, I believe, which is wrong

Reply:
I think it is already out side the dataset version migration. i have checked on based PR.